core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -34,6 +34,8 @@ private CatalogProperties() {
   public static final String HIVE_CLIENT_POOL_SIZE = "clients";
   public static final int HIVE_CLIENT_POOL_SIZE_DEFAULT = 2;
 
+  public static final String HADOOP_SUPPRESS_IOEXCEPTION = "suppress-io-exception";
```

Comment:
Since this is specific to Hadoop, let's move it into `HadoopCatalog`. I don't think the other catalogs would need this because they do not list using FS operations. A private constant in `HadoopCatalog` should be fine and we can always expose it later if we need to.